### PR TITLE
Message auto-capitalization option in character setup

### PR DESCRIPTION
**Pull request by tloy-x**
[FEATURE]

## About the PR 
This PR adds an option in character setup for auto-capitalization of sent chat messages. It is off by default. When enabled, the first letter of your chat messages will be capitalized. Some cool parsing is used to check if it is a radio message or not, and to chop off whitespace at the start of a message. Also adds a "cancel" button to the input box when using the radio hotkey.

## Why's this needed? 
Makes chat messages look more "professional". I just got tired of forgetting to press shift and having my message be all lowercase.

## Changelog

```
(u)tloy-x:
(*)Added option in character setup for auto-capitalization of sent chat messages
(+)Added "cancel" button to input box when using the radio hotkey
```

### CODE REVIEW

File: code/procs/mobprocs/chatprocs.dm [hunk 1]
```diff
@@ -694,6 +728,16 @@
...
 
+/mob/verb/toggle_auto_capitalization()
+	set desc = "Toggles auto capitalization of chat messages"
+	set name = "Toggle Auto Capitalization"
+
+	if (!usr.client)
+		return
+
+	usr.client.preferences.auto_capitalization = !usr.client.preferences.auto_capitalization
+	boutput(usr, "<span class='notice'>[usr.client.preferences.flying_chat_hidden ? "No longer": "Now"] auto capitalizing messages.</span>")
```
Comment (ZeWaka) [inline code/procs/mobprocs/chatprocs.dm:739]: wrong var access

Comment (tloy-x) [reply code/procs/mobprocs/chatprocs.dm]: Fixed

File: code/procs/mobprocs/chatprocs.dm [hunk 2]
```diff
@@ -7,6 +7,30 @@
 /mob/verb/say_verb(message as text)
 	set name = "say"
 	//&& !src.client.holder
+
+	if (!client.preferences.auto_capitalization)
```
Comment (pali6) [inline code/procs/mobprocs/chatprocs.dm:11]: Why is this negated? Shouldn't the code trigger when auto_cpitalization is *on*?

Comment (tloy-x) [reply code/procs/mobprocs/chatprocs.dm:11]: You’re right, it should. I think I screwed up the definition of auto_capitalization when I first wrote it. Fixing now.

**Pull request merged**